pynicotine/slskmessages.py

```diff
@@ -420,40 +428,41 @@ class Login(ServerMessage):
     established. Server responds with the greeting message. """
 
     def __init__(self, username=None, passwd=None, version=None):
+        # TODO: use _str(). also needs to be changed downstream, so defer change for now.
         self.username = username
         self.passwd = passwd
         self.version = version
         self.ip = None
 
-    def makeNetworkMessage(self):
+    def __repr__(self):
+        return f"Login({self.username}, {self.passwd}, {self.version}, {self.ip})"
 
-        m = hashlib.md5()
-        m.update(self.username+self.passwd)
-        md5hash = m.hexdigest()
+    def makeNetworkMessage(self):
+        payload = self.username + self.passwd
+        md5hash = hashlib.md5(payload.encode()).hexdigest()
         message = self.packObject(self.username) + self.packObject(self.passwd) + self.packObject(self.version) + self.packObject(md5hash) + self.packObject(17)
-        # print message.__repr__()
         return message
 
     def parseNetworkMessage(self, message):
-        pos, self.success = 1, ord(message[0])
+        pos, self.success = 1, (ord(message[0]) if isinstance(message[0], str) else message[0])
         if not self.success:
-            pos, self.reason = self.getObject(message, types.StringType, pos)
+            pos, self.reason = self.getObject(message, bytes, pos)
 
         else:
-            pos, self.banner = self.getObject(message, types.StringType, pos)
+            pos, self.banner = self.getObject(message, bytes, pos)
         if len(message[pos:]) > 0:
             try:
                 import socket
-                pos, self.ip = pos+4, socket.inet_ntoa(self.strrev(message[pos:pos+4]))
+                pos, self.ip = pos+4, socket.inet_ntoa(message[pos+4:pos:-1])
```

Comment:
I don't think these do quite the same thing:
```
> "1234567890"[6:2:-1]
'7654'
> 1234567890"[2:6][::-1]
'6543'

Reply:
Oops, indeed. Fixed in b0783e1.